internal/utils/comparators/deployments/comparator.go

```diff
@@ -8,8 +8,10 @@ import (
 	apps "k8s.io/api/apps/v1"
 )
 
+type DeploymentComparator func(current apps.Deployment, desired apps.Deployment) (equal bool, reason string)
+
 // AreSame compares deployments for equality and return true equal otherwise false
-func AreSame(current *apps.Deployment, desired *apps.Deployment) (bool, string) {
+func AreSame(current *apps.Deployment, desired *apps.Deployment, additional ...DeploymentComparator) (bool, string) {
```

Comment:
As far as I can see this is currently called from exactly one place. I think it would be much simpler to just do the additional comparison here instead of adding an additional "comparison callback" that moves the comparison somewhere else.

Same applies to DaemonSet, as far as I can see.

Reply:
It is only separated because its a general use function.  Granted its only used in one place but in the interest of proper design it makes sense IMO to leave it separated.  Though...if its not populated or expected to be populated then it evaluates to success either way